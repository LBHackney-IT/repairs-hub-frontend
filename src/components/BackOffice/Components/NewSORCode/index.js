import { TextArea, TextInput } from '../../../Form'

const NewSORCode = ({
  sorCode,
  handleRemoveSORCode,
  handleSORCodeFieldChange,
  errors,
}) => {
  return (
    <>
      <div id="new-sor-code-container">
        <div
          className="govuk-form-group lbh-form-group"
          id="sor-code-input-grid-area "
        >
          <TextInput
            label="SOR code"
            placeholder="Example: 4896830H"
            name="sor-code-input"
            value={sorCode.code}
            onChange={(e) => handleSORCodeFieldChange(e, sorCode.id, 'code')}
            error={errors?.code && { message: 'Please enter SOR code' }}
          />
        </div>
        <div
          className="govuk-form-group lbh-form-group"
          id="sor-code-cost-input-grid-area"
        >
          <TextInput
            label="Cost (Â£)"
            placeholder="Example: 10.47"
            name="sor-code-cost-input"
            value={sorCode.cost}
            type="number"
            min={0}
            onChange={(e) => handleSORCodeFieldChange(e, sorCode.id, 'cost')}
            error={errors?.cost && { message: 'Please enter cost value' }}
          />
        </div>
        <div
          className="govuk-form-group lbh-form-group"
          id="sor-code-smv-input-grid-area"
        >
          <TextInput
            label="Standard Minute Value"
            placeholder="Example: 29"
            name="sor-code-smv-input"
            value={sorCode.standardMinuteValue}
            type="number"
            min={0}
            onChange={(e) =>
              handleSORCodeFieldChange(e, sorCode.id, 'standardMinuteValue')
            }
            error={
              errors?.standardMinuteValue && {
                message: 'Please enter SMV value',
              }
            }
          />
        </div>
        <div id="sor-code-remove-grid-area" style={{ textAlign: 'center' }}>
          <a
            className="lbh-link"
            href="#"
            onClick={(e) => handleRemoveSORCode(e, sorCode)}
          >
            Remove SOR code
          </a>
        </div>
        <div
          className="govuk-form-group lbh-form-group"
          id="sor-code-short-desc-input-grid-area"
        >
          <TextArea
            label="Short description"
            placeholder="Example: LH Gas Carcass LGSR inc cooker"
            name="sor-code-short-desc-input"
            value={sorCode.shortDescription}
            rows="1"
            onChange={(e) =>
              handleSORCodeFieldChange(e, sorCode.id, 'shortDescription')
            }
            error={
              errors?.shortDescription && {
                message: 'Please enter short description',
              }
            }
          />
        </div>
        <div
          className="govuk-form-group lbh-form-group"
          id="sor-code-long-desc-input-grid-area"
        >
          <TextArea
            label="Long description"
            placeholder="Example: LH Gas Carcass test. Test internal gas pipework for soundness from meter to all appliances. Visual check cooker or any other gas appliances, excludes gas fire. Issue LGSR Landlord gas safety record..."
            name="sor-code-long-desc-input"
            value={sorCode.longDescription}
            rows="2"
            onChange={(e) =>
              handleSORCodeFieldChange(e, sorCode.id, 'longDescription')
            }
            error={
              errors?.longDescription && {
                message: 'Please enter long description',
              }
            }
          />
        </div>
      </div>
      <hr />
    </>
  )
}

export default NewSORCode
