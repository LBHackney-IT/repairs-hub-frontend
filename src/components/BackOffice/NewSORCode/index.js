const NewSORCode = ({ sorCode, handleRemoveSORCode, handleSORCodeFieldChange }) => {

    return (
        <>
            <div className="govuk-form-group lbh-form-group">
                <label className="govuk-label lbh-label" htmlFor="sor-code-input">
                    SOR code <span id="input-with-hint-text-hint" className="govuk-hint lbh-hint">Example: 4896830H</span>
                </label>

                <input
                    className="govuk-select lbh-select"
                    name="sor-code-input"
                    type="text"
                    value={sorCode.sorCode}
                    id='sor-code-input'
                    onChange={(e) => handleSORCodeFieldChange(e, sorCode.id, 'sorCode')}
                />
            </div>
            <div className="govuk-form-group lbh-form-group">
                <label className="govuk-label lbh-label" htmlFor="sor-code-cost-input">
                    Cost (Â£) <span id="input-with-hint-text-hint" className="govuk-hint lbh-hint">Example: 10.47</span>
                </label>

                <input
                    className="govuk-select lbh-select"
                    name="sor-code-cost-input"
                    type="text"
                    value={sorCode.cost}
                    id='sor-code-cost-input'
                    onChange={(e) => handleSORCodeFieldChange(e, sorCode.id, 'cost')}
                />
            </div>
            <button onClick={() => handleRemoveSORCode(sorCode)}>
                Remove
            </button>
        </>
    )
}

export default NewSORCode