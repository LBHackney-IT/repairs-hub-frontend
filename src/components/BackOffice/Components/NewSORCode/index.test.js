import SorCode from '@/root/src/models/sorCode'
import { render } from '@testing-library/react'
import NewSORCode from '.'

describe('NewSORCode component', () => {
  it('should render the component', async () => {
    const sorCode = new SorCode(1, 'ASD123YO', 25.50, 47, 'Short Description', 'Long Description')
    const { asFragment } = render(<NewSORCode 
      key={sorCode.id}
      sorCode={sorCode}
      handleRemoveSORCode={jest.fn()}
      handleSORCodeFieldChange={jest.fn()}/>)

    expect(asFragment()).toMatchSnapshot()
  })
})
