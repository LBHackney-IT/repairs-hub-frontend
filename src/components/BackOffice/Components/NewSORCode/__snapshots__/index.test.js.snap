// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewSORCode component should render the component 1`] = `
<DocumentFragment>
  <div
    id="new-sor-code-container"
  >
    <div
      class="govuk-form-group lbh-form-group"
      id="sor-code-input-grid-area "
    >
      <div
        class="govuk-form-group lbh-form-group"
        id="sor-code-cost-code-form-group"
      >
        <label
          class="govuk-label lbh-label"
          for="sor-code-cost-code"
        >
          SOR code 
        </label>
        <input
          class="govuk-input lbh-input undefined"
          data-testid="sor-code-cost-code"
          id="sor-code-cost-code"
          name="sor-code-cost-code"
          placeholder="Example: 4896830H"
          type="text"
          value="ASD123YO"
        />
      </div>
    </div>
    <div
      class="govuk-form-group lbh-form-group"
      id="sor-code-cost-input-grid-area"
    >
      <div
        class="govuk-form-group lbh-form-group"
        id="sor-code-cost-input-form-group"
      >
        <label
          class="govuk-label lbh-label"
          for="sor-code-cost-input"
        >
          Cost (Â£) 
        </label>
        <input
          class="govuk-input lbh-input undefined"
          data-testid="sor-code-cost-input"
          id="sor-code-cost-input"
          min="0"
          name="sor-code-cost-input"
          placeholder="Example: 10.47"
          type="number"
          value="25.5"
        />
      </div>
    </div>
    <div
      class="govuk-form-group lbh-form-group"
      id="sor-code-smv-input-grid-area"
    >
      <div
        class="govuk-form-group lbh-form-group"
        id="sor-code-smv-input-form-group"
      >
        <label
          class="govuk-label lbh-label"
          for="sor-code-smv-input"
        >
          Standard Minute Value 
        </label>
        <input
          class="govuk-input lbh-input undefined"
          data-testid="sor-code-smv-input"
          id="sor-code-smv-input"
          min="0"
          name="sor-code-smv-input"
          placeholder="Example: 29"
          type="number"
          value="47"
        />
      </div>
    </div>
    <div
      id="sor-code-remove-grid-area"
      style="text-align: center;"
    >
      <a
        class="lbh-link"
        href="#"
      >
        Remove SOR code
      </a>
    </div>
    <div
      class="govuk-form-group lbh-form-group"
      id="sor-code-short-desc-input-grid-area"
    >
      <div
        class="govuk-form-group lbh-form-group"
        id="sor-code-short-desc-input-form-group"
      >
        <label
          class="govuk-label lbh-label"
          for="sor-code-short-desc-input"
        >
          Short description 
        </label>
        <textarea
          aria-describedby="sor-code-short-desc-input-hint sor-code-short-desc-input-error"
          class="govuk-textarea lbh-textarea"
          data-testid="sor-code-short-desc-input"
          id="sor-code-short-desc-input"
          name="sor-code-short-desc-input"
          placeholder="Example: LH Gas Carcass LGSR inc cooker"
          rows="1"
        >
          Short Description
        </textarea>
      </div>
    </div>
    <div
      class="govuk-form-group lbh-form-group"
      id="sor-code-long-desc-input-grid-area"
    >
      <div
        class="govuk-form-group lbh-form-group"
        id="sor-code-long-desc-input-form-group"
      >
        <label
          class="govuk-label lbh-label"
          for="sor-code-long-desc-input"
        >
          Long description 
        </label>
        <textarea
          aria-describedby="sor-code-long-desc-input-hint sor-code-long-desc-input-error"
          class="govuk-textarea lbh-textarea"
          data-testid="sor-code-long-desc-input"
          id="sor-code-long-desc-input"
          name="sor-code-long-desc-input"
          placeholder="Example: LH Gas Carcass test. Test internal gas pipework for soundness from meter to all appliances. Visual check cooker or any other gas appliances, excludes gas fire. Issue LGSR Landlord gas safety record..."
          rows="2"
        >
          Long Description
        </textarea>
      </div>
    </div>
  </div>
  <hr />
</DocumentFragment>
`;
